snapcraft/projects.py

```diff
@@ -822,6 +822,13 @@ def get_build_for_arch_triplet(self) -> Optional[str]:
 
         return None
 
+    def get_component_names(self) -> Optional[List[str]]:
```

Comment:
Any reason for this rather than just returning an empty `List[str]` if there are no components?

Reply:
I think that's a better idea, let me implement it that way.